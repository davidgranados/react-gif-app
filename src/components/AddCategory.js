import React, {useState} from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({onAddCategory}) => {
  const [newCategory, setNewCategory] = useState('');
  const handleNewCategoryChange = ({target}) => {
    setNewCategory(target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (newCategory.trim().length > 2) {
      onAddCategory(newCategory);
      setNewCategory('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" onChange={handleNewCategoryChange} value={newCategory}/>
      <button type="submit">Add New Category</button>
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired
};

export default AddCategory;
