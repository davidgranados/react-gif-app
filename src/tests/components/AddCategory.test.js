import React from "react";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Tests on AddCategory component", function () {
  const wrapper = shallow(<AddCategory onAddCategory={() => {}} />);
  const event = {
    target: {
      value: "Hola mundo",
    },
  };
  it("component should show correctly", function () {
    expect(wrapper).toMatchSnapshot();
  });
  it("should change newCategory state", function () {
    expect(wrapper.find("input").props().value).toEqual("");
    wrapper.find("input").simulate("change", event);
    expect(wrapper.find("input").props().value).toEqual(event.target.value);
  });
});
