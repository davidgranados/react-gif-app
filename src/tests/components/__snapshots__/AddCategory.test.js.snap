// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AddCategory component component should show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add New Category
  </button>
</form>
`;
